### PR TITLE
feat: support use vscode variables in settings

### PR DESCRIPTION
**Pull request by njzydark**
https://github.com/nix-community/nixd/blob/main/nixd/docs/configuration.md#where-to-place-the-configuration

According to the above document, when defining the expr of options, only absolute paths can be used. This is very inconvenient for configuring across machines. Therefore, this PR introduces the [vscode-variables](https://www.npmjs.com/package/vscode-variables) package to support using `${workspaceFolder}` in configuration files to define paths.

before

```json
{
    "nixd": {
      "options": {
        "nixpkgs": {
          "expr": "import <nixpkgs> { }"
        },
        "nix-darwin": {
          "expr": "(builtins.getFlake \"/absolute/path/to/flake\").darwinConfigurations.default.options"
        }
      }
    }
}
```

after

```json
{
    "nixd": {
      "options": {
        "nixpkgs": {
          "expr": "import <nixpkgs> { }"
        },
        "nix-darwin": {
          "expr": "(builtins.getFlake \"${workspaceFolder}/flake\").darwinConfigurations.default.options"
        }
      }
    }
}
```

### CODE REVIEW

File: src/global.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+declare module "vscode-variables" {
+  export default function variables(
+    string: string,
+    recursive?: boolean,
+  ): string;
```
Comment (jnoortheen) [inline src/global.d.ts:5]: do we need to export these types? is this needed by vscode ?

File: package.json [hunk 2]
```diff
@@ -150,6 +150,7 @@
   },
   "dependencies": {
     "command-exists": "^1.2.9",
-    "vscode-languageclient": "^9.0.0"
+    "vscode-languageclient": "^9.0.0",
+    "vscode-variables": "^0.1.3"
   }
```
Comment (jnoortheen) [inline package.json:155]: could you document this feature ? to the readme concisely perhaps

File: src/global.d.ts (hunk 1 above)
Comment (njzydark) [reply src/global.d.ts:5]: This type definition is mainly for solving type checking issues.

File: README.md [hunk 3]
```diff
@@ -73,6 +73,10 @@ Pass settings to the language server via the `serverSettings` option.
                 },
                 "home-manager": {
                     "expr": "(builtins.getFlake \"/absolute/path/to/flake\").homeConfigurations.<name>.options"
+                },
+                // Tip: use ${workspaceFolder} variable to define path
```
Comment (njzydark) [inline README.md:77]: describe this feature here

**Review by jnoortheen — APPROVED** [review]

**Comment by jnoortheen** [comment]
Thanks for the contribution @njzydark !

**Pull request merged**